oidc_auth/authentication.py

```diff
@@ -18,11 +18,27 @@
 
 def get_user_by_id(request, id_token):
     User = get_user_model()
-    try:
-        user = User.objects.get_by_natural_key(id_token.get('sub'))
-    except User.DoesNotExist:
-        msg = _('Invalid Authorization header. User not found.')
-        raise AuthenticationFailed(msg)
+    if api_settings.CREATE_USER:
+        email = id_token.get('email', None)
+        first_name = id_token.get('given_name', '')
+        last_name = id_token.get('family_name', '')
+        if email:
+            user, created = User.objects.get_or_create(username=id_token.get('sub').lower(),
+                                                       defaults={'first_name': first_name,
+                                                                 'last_name': last_name,
+                                                                 'email': email,
+                                                                 })
+        else:
+            user, created = User.objects.get_or_create(username=id_token.get('sub'.lower()),
+                                                       defaults={'first_name': first_name,
+                                                                 'last_name': last_name,
+                                                                 })
+    else:
+        try:
+            user = User.objects.get_by_natural_key(id_token.get('sub'))
+        except User.DoesNotExist:
+            msg = _('Invalid Authorization header. User not found.')
+            raise AuthenticationFailed(msg)
```

Comment:
It would be nice if there were some tests for this, since this is quite some modification.

Reply:
see my comments on testing below - thanks